js/src/ui/TxList/TxRow/txRow.js

```diff
@@ -137,11 +147,98 @@ class TxRow extends Component {
     return (
       <td className={ styles.timestamp }>
         <div>{ blockNumber && block ? moment(block.timestamp).fromNow() : null }</div>
-        <div>{ blockNumber ? _blockNumber.toFormat() : 'Pending' }</div>
+        <div>{ blockNumber ? _blockNumber.toFormat() : this.renderCancelToggle() }</div>
       </td>
     );
   }
 
+  renderCancelToggle () {
+    const { canceled, editing, isCancelOpen, isEditOpen } = this.state;
+
+    if (canceled) {
+      return (
+        <div className={ styles.pending }>
+          <FormattedMessage
+            id='ui.txList.txRow.canceled'
+            defaultMessage='CANCELED'
+          />
+        </div>
+      );
+    }
+
+    if (editing) {
+      return (
+        <div className={ styles.pending }>
+          <FormattedMessage
+            id='ui.txList.txRow.editing'
+            defaultMessage='EDITING'
+          />
+        </div>
+      );
+    }
+
+    if (!isCancelOpen && !isEditOpen) {
+      return (
+        <div className={ styles.pending }>
+          <span>
+            <FormattedMessage
+              id='ui.txList.txRow.time'
+              defaultMessage='{when}'
+              values={ { when: this.getCondition() } }
+            />
+          </span>
+          <div>
+            <FormattedMessage
+              id='ui.txList.txRow.scheduled'
+              defaultMessage='SCHEDULED'
+            />
+          </div>
+          <a onClick={ this.setEdit }>
+            <FormattedMessage
+              id='ui.txList.txRow.edit'
+              defaultMessage='EDIT'
+            />
+          </a>
+          <span>{' | '}</span>
+          <a onClick={ this.setCancel }>
+            <FormattedMessage
+              id='ui.txList.txRow.cancel'
+              defaultMessage='CANCEL'
+            />
+          </a>
+        </div>
+      );
+    }
+
+    return (
+      <div className={ styles.pending }>
+        <div />
+        <div>
+          <FormattedMessage
+            id='ui.txList.txRow.verify'
+            defaultMessage='ARE YOU SURE?'
+          />
+        </div>
+        <a onClick={ (isCancelOpen) ? this.cancelTransaction : this.editTransaction }>
+          <FormattedMessage
+            id='ui.txList.txRow.verify.cancelEdit'
+            defaultMessage='{ which }'
+            values={ {
+              which: `${(isCancelOpen) ? 'Cancel' : 'Edit'}`
```

Comment:
We should get a big red/yellow warning for transaction cancellation since it's not reliable as mentioned in RPC docs.

There are two separate cases:
1. Scheduled transactions and transactions that were not propagated to other peers - such transactions can be safely canceled, it will be removed from our local queue and never propagated to the network so it's fine.
2. Other transactions - those transactions cannot be simply "canceled" since they are already floating around a network and there is a very high chance that other peers have accepted the transaction and are relaying it further.
For those transactions we should display a warning (but still allow that) and offer another "cancelation" mechanism decribed here: https://github.com/paritytech/parity/issues/4519, so: Increase gas price, remove data, lower gas and submit transaction again.

When editing transaction we should also differentiate those two cases. (1) is safe to edit, but (2) should always increase the gas price when edited (it increases the chance of actually replacing the transaction that was already propagated).